### ISSUE TITLE
[GUI] Editting a finded patient information clear my find result

### ISSUE THREAD

**Issue by mrnobody0505** [mrnobody0505/pe#3]
Step to reproduce:
1) `find john` will give something like this:

![image.png](https://raw.githubusercontent.com/mrnobody0505/pe/main/files/abe1b6ba-41bd-4acc-a7d9-783cd741c5aa.png)

2) Then, `edit 2 n/ Pete Does`

![image.png](https://raw.githubusercontent.com/mrnobody0505/pe/main/files/76467b1e-e032-40a1-a4c6-59a8698d9b02.png)

3) Then, press command for execution.

Expected output: The find list after step 1 remains with the updated name of the patient.

Actual output: The whole list of all patient return

![image.png](https://raw.githubusercontent.com/mrnobody0505/pe/main/files/084b12a5-41f8-43ea-951a-6c1acbe805b7.png)

As an user who is focusing on the list of found patients, I want the page to remain in order to operate on the returned records from find command. However, returning the entire list might be troublesome for the user to enter find command again to see the edited field. If in case I want to check if the patient from the find command is editted correctly, I have to scroll all the way through the entire list to check, which is really inconvenient. Hence, I would give this a medium severity.

**Comment by nus-se-script** [comment]
# Team's Response
No details provided by team.

## The 'Original' Bug

[The team marked this bug as a duplicate of the following bug]

># Editing a Field of a Patient Clears Find Results‬ ‭
>
>**Description:‬**
>‭ When editing a field of a patient found via the‬‭ find‬‭ command, the patient list unexpectedly‬ resets, displaying the entire list rather than retaining the filtered search results. This disrupts the‬ user's workflow, requiring them to re-enter the‬‭ find‬‭ command to continue working with the‬ targeted records.‬
>
>**Steps to Reproduce:‬**
>‭ 1.‬‭ Execute a find command, e.g.,‬‭ find n/David to display‬‭ a filtered list of patients.‬
>‭
>○
>‬‭ Screenshot:‬‭ 
>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/d9ef79f8-2b9a-4210-9db1-c961d04e8ff2.png)
>
>‭ 2.‬‭ Use an edit command, to update the‬
>‭ patient's details.‬
>
>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/a93c3491-7927-45ca-9ad1-e37d5a807643.png)
>
>‭ 3.‬‭ Press Enter to execute the edit command.‬
>
>**‭ Expected Result:‬**
>‭ The application should retain the filtered‬‭ find‬‭ results‬‭ in the patient list, showing only the‬ matched patients, with the edited field updated accordingly.‬
>
>‭ **Actual Result:‬**
>‭ The patient list resets to show all patients rather than retaining the filtered results.‬
>‭
>●
>‬‭ Screenshot:‬‭ (Add screenshot showing entire list after‬‭ edit)‬
>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/9f3e8fab-5b8a-4c22-ba8e-977f76445774.png)
>
>**‭ Severity:‬**
>‭Severity Level:‬‭ Medium‬
>‭
>**Rationale:**‬‭ This behavior interrupts the user experience,‬‭ requiring users to re-enter the‬
>‭ find‬‭ command to continue working with the target record, users may not remember all the fields he wants to change in one go. Furthermore, user may want to see the change being reflected. He will have to find again to see the change, causing a lot of inconvenience.‬‭ 
>‭
>‭
>
>
>
>
>
>
><hr>
><sub>[original: nus-cs2103-AY2425S1/pe-interim#2657] [original labels: severity.Medium type.FeatureFlaw]</sub>
>
>
>

## Their Response to the 'Original' Bug

[This is the team's response to the above 'original' bug]

>Thank you for the bug report!
>
>This is actually a deliberate design choice and intended behaviour as detailed in our User Guide, as per the last bullet point in the screnshot below.
>
>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/8ab15b17-fe2b-46c3-8b95-e04ce3746936.png)
>
>While it may initially seem inconvenient for the patient list to reset when editing a field after a find command, there are several reasons why this behavior would be beneficial and preferable to retaining the filtered results that led us to make this design decision:
>
>1. Firstly, a direct benefit would be to ensure clear and consistent state management. Resetting the list ensures that the app maintains a clear and consistent state after an edit. As such, the user interacts with the application in a straightforward manner where the system always starts from a known, consistent state (i.e., showing the entire patient list). This avoids potential confusion caused by trying to track changes or filter results based on outdated or altered data.
>
>2. More importantly, this prevents data discrepancies. If the filtered list was retained post-edit, there could be inconsistencies between the displayed results and the underlying data, especially if the changes made during the edit might affect the criteria used by the find command. Resetting the list ensures the user is working with the most accurate and up-to-date dataset, without inadvertently displaying outdated or mismatched results.
>
>    While this could be rectified by reapplying the filter logic after each edit to retain the updated find command results, this would result in significant performance overhead, potentially slowing down the system, especially when dealing with a large patient database. As such, a reset of the list helps improve performance by avoiding unnecessary complexity in repeated querying and re-rendering or filtering data after each edit.
>
>3. From a UX perspective, this also provides clearer context for users. When the patient list resets, the user can easily identify that they have moved away from their filtered search results, avoiding any confusion about which data is being worked on. This might be particularly important for less tech-savvy users who could be unsure whether the correct patient data was updated if the list remained filtered. 
>
>    This also helps to avoid inaccurate data updates. Retaining the filtered list after an edit could lead to scenarios where users mistakenly assume they are editing the correct patient when, in fact, their filtered search results might have changed due to the edit. For example, if the list has been set to be sorted the patients by alphabetical order of their names using the `list` command. Changing a person's name to start with 'Z' from 'A' would move him from the top to the bottom of the list. Should the user not realise this, they may make inaccurate data updates on the new patient at the top of the list. On the other hand, resetting the list helps mitigate this risk by ensuring that the user is not working with a misleading or potentially incomplete dataset.
>
>
>Nevertheless, we do acknowledge that the behaviour of the patient list to show all patients might be frustrating to users who want to mass edit several patients' information at one go. 
>
>Furthermore, we also do currently have status messages that show the new information of the edited contact, shown in the screenshot below, for users to see the change being reflected rather than having to find again to see the change.
>
>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/5b075dfa-2cb4-4edd-80b9-9d9de2ff3593.png)
>
>This information would allow the user to resolve the inconvenience of the patient list being refreshed. The user can execute the `find` command to search for the patient's name again by viewing the name in the status message which is shown immediately after the user enters the `edit` command, and continue to make further edits.  
>
>Given that this only causes "minor inconvenience" for a rather specific use case, we are classifying this as low severity
>
>With the limited scope and time that we had to develop MedConnect, our team felt that while this design implementation would have been ideal to maximise user experience for the current version of the app. Future versions of the app could have commands to reapply the filters or include functionality to toggle the behaviour between refreshing and persisting under User Preferences.
>
>Improving our app to retain the filtered patient list after editing a patient is definitely in our future considerations to improve the user experience of MedConnect. Hence, we have reclassified this issue to be not in scope.
# Items for the Tester to Verify
## :question: Issue duplicate status

Team chose to mark this issue as a duplicate of another issue (as explained in the _**Team's response**_ above)

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
## :question: Issue response

Team chose [`response.NotInScope`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
## :question: Issue severity

Team chose [`severity.Low`]
Originally [`severity.Medium`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
